src/invenio-records-js/controllers/InvenioRecordsCtrl.js

```diff
@@ -155,78 +155,83 @@ function InvenioRecordsCtrl($scope, $rootScope, $q, $window, $location,
+    * @param {String} method - The method (POST, PUT, DELETE).
+    */
+  function makeActionRequest(type, method) {
+    var _data = cleanData();
+    return InvenioRecordsAPI.request({
+      url: vm.invenioRecordsEndpoints[type],
+      method: (method || 'PUT').toUpperCase(),
+      data: _data,
+      headers: vm.invenioRecordsArgs.headers || {}
+    });
+  }
 
-        if (!angular.isUndefined(vm.invenioRecordsEndpoints[type])) {
-          InvenioRecordsAPI.request({
-            url: vm.invenioRecordsEndpoints[type],
-            method: (method || 'PUT').toUpperCase(),
-            data: {
-              metadata: _data
-            },
-            headers: vm.invenioRecordsArgs.headers || {}
-          }).then(
-            successCallback,
-            errorCallback
-          );
-        } else {
-          errorCallback({
-            type: 'danger',
-            data: {
-              message: 'The action type is not supported.'
-            }
-          });
-        }
-      },
-      errorCallback);
+  /**
+    * Handle the redirection after a success action if needed
+    * @memberof InvenioRecordsCtrl
+    * @function handleActionRedirection
+    * @param {String} redirect_path - The url to redirect on success.
+    */
+  function handleActionRedirection(redirect_path) {
+    // Redirect if defined
+    if (!angular.isUndefined(redirect_path) && redirect_path !== '') {
+      // Redirect to new location
+      var _url = redirect_path;
+      if (redirect_path.substr(0, 1) !== '/' && redirect_path.substr(0, 4) !== 'http') {
+        // Find the url
+        _url = vm.invenioRecordsEndpoints[redirect_path];
+      }
+      $window.location.href = _url;
+    }
   }
 
   /**
     * Action handler
     * @memberof InvenioRecordsCtrl
     * @function invenioRecordsHandler
-    * @param {string} type - The action key from ``links`` object.
-    * @param {string} method - The request method type i.e. GET, POST, PUT.
-    * @param {string} redirect_path - The url to redirect on success.
+    * @param {Array} actions - Actions array [[type, method]].
+    * @param {String} redirect_path - The url to redirect on success.
     */
-  function invenioRecordsHandler(type, method, redirect_path) {
+  function invenioRecordsHandler(actions, redirect_path) {
 
+    var _actions = (typeof(actions[0]) === 'string') ? [actions] : actions;
     /**
       * After a successful request
       * @memberof invenioRecordsHandler
       * @function actionSuccessful
-      * @param {Object} response - The action request response.
+      * @param {Object} responses - The promise action request responses.
       */
-    function actionSuccessful(response) {
+    function actionSuccessful(responses) {
+      // NOTE: We keep only the response of the last action!!
+      var response = responses[responses.length - 1] || {};
+
       $rootScope.$broadcast('invenio.records.alert', {
         type: 'success',
         data: response.data,
-        action: type,
```

Comment:
Passing the action type here, used to help with displaying custom messages, if there was no `message` key in the `response.data`. IMHO it should stay like `action: _actions`.